subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/resolve/GenerateGraphTask.groovy

```diff
@@ -13,104 +13,212 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+//file:noinspection GrMethodMayBeStatic
 
 package org.gradle.integtests.fixtures.resolve
 
-import org.gradle.api.artifacts.Configuration
-import org.gradle.api.artifacts.ResolvedDependency
+import org.gradle.api.DefaultTask
+import org.gradle.api.artifacts.ArtifactCollection
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
+import org.gradle.api.artifacts.component.ProjectComponentIdentifier
+import org.gradle.api.artifacts.result.ComponentSelectionCause
+import org.gradle.api.artifacts.result.DependencyResult
+import org.gradle.api.artifacts.result.ResolvedArtifactResult
+import org.gradle.api.artifacts.result.ResolvedComponentResult
+import org.gradle.api.artifacts.result.ResolvedDependencyResult
+import org.gradle.api.artifacts.result.ResolvedVariantResult
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.attributes.AttributeContainer
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileCollection
+import org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.ComponentSelectionReasonInternal
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
+import org.gradle.util.Path
+
+/**
+ * This task writes information about a resolved graph to a file in a flat format (one line per item).
+ *
+ * It exists in order to compare the result of resolution to an expectation; and to ensure that the multiple
+ * ways of accessing artifacts and files from a resolved graph are consistent.
+ *
+ * This task intentionally does <strong>NOT</strong> test the {@link org.gradle.api.artifacts.ResolvedConfiguration ResolvedConfiguration} API,
+ * which ought to be considered legacy and deprecated for internal use.  That type's behavior will be verified elsewhere.
+ * This is meant to be Configuration Cache compatible, so it does not test the {@link org.gradle.api.artifacts.result.ResolutionResult ResolutionResult} API
+ * yet - when that type is made CC compatible, this task should be updated to test it as well.
```

Comment:
I think ResolvedComponentResult is the CC-compatible type.   I don't know if we're going to make ResolutionResult compatible ever.

Reply:
So the `DependencyResult` set contained in the root `ResolvedComponentResult` present in the `ResolutionResult` is already compared when the graph is checked, prior to checking files and artifacts.  It is indeed CC-compatible and stored as the `rootComponent` property.

I went through and added the remaining artifact/file collections to the verification just to make this completely comprehensive.